hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bucket/HoodieSparkConsistentBucketIndex.java

```diff
@@ -271,8 +327,114 @@ public Option<HoodieRecordLocation> getRecordLocation(HoodieKey key) {
+      Path metadataPartitionPath = FSUtils.getPartitionPath(table.getMetaClient().getHashingMetadataPath(), partition);
+      Path metadataFilePath = new Path(metadataPartitionPath, hoodieInstant + HASHING_METADATA_FILE_SUFFIX);
+      try {
+        if (table.getMetaClient().getFs().exists(metadataFilePath)) {
+          createCommitMarker(table, metadataFilePath, metadataPartitionPath);
+        }
+      } catch (IOException e) {
+        throw new HoodieIOException("exception while committing hashing metadata for path " + metadataFilePath, e);
+      }
+    });
+  }
+
+  /***
+   * Create commit marker corresponding to hashing metadata file after post commit clustering operation
+   * @param table
+   * @param fileStatus
+   * @param partitionPath
+   * @throws IOException
+   */
+  private static void createCommitMarker(HoodieTable table, Path fileStatus, Path partitionPath) throws IOException {
+    HoodieWrapperFileSystem fs = table.getMetaClient().getFs();
+    Path fullPath = new Path(partitionPath, getTimestampFromFile(fileStatus.getName()) + HASHING_METADATA_COMMIT_FILE_SUFFIX);
+    if (fs.exists(fullPath)) {
+      return;
+    }
+    String metadata = "";
+    FileIOUtils.createFileInPath(fs, fullPath, Option.of(metadata.getBytes()));
+  }
+
+  /***
+   * Load consistent hashing metadata from given file
+   * @param table
+   * @param metaFile
+   * @return
+   */
+  private static Option<HoodieConsistentHashingMetadata> loadMetadataFromGivenFile(HoodieTable table, FileStatus metaFile) {
+    try {
+      if (metaFile == null) {
+        return Option.empty();
+      }
+      byte[] content = FileIOUtils.readAsByteArray(table.getMetaClient().getFs().open(metaFile.getPath()));
+      return Option.of(HoodieConsistentHashingMetadata.fromBytes(content));
+    } catch (FileNotFoundException e) {
+      return Option.empty();
+    } catch (IOException e) {
+      LOG.error("Error when loading hashing metadata, for path: " + metaFile.getPath().getName(), e);
+      throw new HoodieIndexException("Error while loading hashing metadata", e);
+    }
+  }
+
+  /***
+   * COMMIT MARKER RECOVERY JOB.
+   * If particular hashing metadta file doesn't have commit marker then there could be a case where clustering is done but post commit marker
+   * creation operation failed. In this case this method will check file group id from consistent hashing metadata against storage base file group ids.
+   * if one of the file group matches then we can conclude that this is the latest metadata file.
+   * Note : we will end up calling this method if there is no marker file and no replace commit on active timeline, if replace commit is not present on
+   * active timeline that means old file group id's before clustering operation got cleaned and only new file group id's  of current clustering operation
+   * are present on the disk.
+   * @param table
```

Comment:
Adds the comment of all parameters and return.

Reply:
sure

Reply:
done